### PR TITLE
KOGITO-5597 Separate native checks

### PR DESCRIPTION
**Pull request by radtriste**
https://issues.redhat.com/browse/KOGITO-5597

Related PRs:
- https://github.com/kiegroup/kogito-runtimes/pull/1482
- https://github.com/kiegroup/optaplanner/pull/1472
- https://github.com/kiegroup/kogito-apps/pull/965
- https://github.com/kiegroup/kogito-examples/pull/818

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* for a <b>pull request build</b> please add comment: <b>Jenkins retest this</b>
* for a <b>specific pull request build</b> please add comment: <b>Jenkins (re)run [optaplanner|apps|examples] tests</b>
* for a <b>full downstream build</b> please add comment: <b>Jenkins run fdb</b>
* for a <b>compile downstream build</b> please add comment: <b>Jenkins run cdb</b>
* for a <b>full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>
* for an <b>upstream build</b> please add comment: <b>Jenkins run upstream</b>
* for a <b>Quarkus LTS check</b> please add comment: <b>Jenkins run LTS</b>
* for a <b>specific Quarkus LTS check</b> please add comment: <b>Jenkins (re)run [optaplanner|apps|examples] LTS</b>
* for a <b>Native check</b> please add comment: <b>Jenkins run native</b>
* for a <b>specific Native LTS check</b> please add comment: <b>Jenkins (re)run [optaplanner|apps|examples] native</b>
</details>

### CODE REVIEW

**Comment by radtriste** [comment]
When running the check, noticed that quickstarts has a problem: https://eng-jenkins-csb-business-automation.apps.ocp4.prod.psi.redhat.com/job/custom/job/tradisso/job/kogito-5597/job/nightly/job/kogito-5597/job/optaplanner-native/

**Comment by radtriste** [comment]
@rsynek implemented yet with normal Jenkins and we can later use the buildchain when it is operational :)

File: .gitignore [hunk 1]
```diff
@@ -4,7 +4,7 @@
 # Eclipse, Netbeans and IntelliJ files
 /.*
 !/.github
-!/.jenkins
+!/.ci
```
Comment (ge0ffrey) [inline .gitignore:7]: I presume this file needs this change too over at optaplanner-quickstarts?
  https://github.com/kiegroup/optaplanner-quickstarts/blob/stable/.gitignore
If it does, would you mind making a PR for that too?

**Review by ge0ffrey — APPROVED** [review]

File: .ci/jenkins/Jenkinsfile [hunk 2]
```diff
@@ -59,8 +59,7 @@ pipeline {
             steps {
                 script {
                     getMavenCommand(kogitoRuntimesRepo)
-                        .skipTests(true)
-                        .withProperty('skipITs', true)
+                        .withProperty('quickly')
```
Comment (ge0ffrey) [inline .ci/jenkins/Jenkinsfile:62]: If this `-Dquickly` is the new way to build quickly, standarized across all KIE projects folowing Quarkus example, that is great. Would you mind doing a quick why-to-use + how-to-use on the zulip optaplanner channel (so the optaplanner team knows) and adjust the README.adoc to mention it exists?

**Review by rsynek — APPROVED** [review]

**Comment by rsynek** [comment]
> When running the check, noticed that quickstarts has a problem: https://eng-jenkins-csb-business-automation.apps.ocp4.prod.psi.redhat.com/job/custom/job/tradisso/job/kogito-5597/job/nightly/job/kogito-5597/job/optaplanner-native/

Caused by https://issues.redhat.com/browse/PLANNER-2489. The workaround is already in place, it should work now.

**Comment by rsynek** [comment]
jenkins rerun optaplanner tests

**Comment by radtriste** [comment]
jenkins rerun optaplanner tests

File: .ci/jenkins/Jenkinsfile (hunk 2 above)
Comment (radtriste) [reply .ci/jenkins/Jenkinsfile:62]: This was taken from Quarkus and I think is only implemented on Kogito/Optaplanner.
Kogito doc was already updated: https://github.com/kiegroup/kogito-runtimes/blob/main/CONTRIBUTING.md#build
I can leave a quick message on Optaplanner Zulip channel about it.

**Comment by radtriste** [comment]
jenkins rerun examples tests

**Review by Ginxo — APPROVED** [review]
LGTM

**Pull request merged**